tools/cli/adminKafkaCommands.go

```diff
@@ -502,23 +501,23 @@ func AdminRereplicate(c *cli.Context) {
 		return
 	}
 	sourceCluster := getRequiredOption(c, FlagSourceCluster)
-	if !c.IsSet(FlagMaxEventID) {
-		ErrorAndExit("End event ID is not defined", nil)
-	}
-	if !c.IsSet(FlagEndEventVersion) {
-		ErrorAndExit("End event version is not defined", nil)
-	}
 
 	adminClient := cFactory.ServerAdminClient(c)
-	endEventID := c.Int64(FlagMaxEventID)
-	endVersion := c.Int64(FlagEndEventVersion)
+	var endEventID, endVersion *int64
+	if c.IsSet(FlagMaxEventID) {
+		endEventID = common.Int64Ptr(c.Int64(FlagMaxEventID) + 1)
+	}
+	if c.IsSet(FlagEndEventVersion) {
+		endVersion = common.Int64Ptr(c.Int64(FlagEndEventVersion))
+	}
 	domainID := getRequiredOption(c, FlagDomainID)
 	wid := getRequiredOption(c, FlagWorkflowID)
 	rid := getRequiredOption(c, FlagRunID)
 	shardID := common.WorkflowIDToHistoryShard(wid, numberOfShards)
 	contextTimeout := defaultResendContextTimeout
 	executionManager := initializeExecutionStore(c, shardID, 0)
```

Comment:
@yux0 Do we have to do the 2dc workflow check in the command? 
Currently the re-replication command should be issued to the target cluster. But when we use re-replication, typically the workflow doesn't exist in target cluster and we have to specify the DB option for the source cluster in order to get and verify the version history. 
I wonder if the admin GetRawHistory handler already catches the case where the workflow is 2dc and returns a bad request error? If so, I guess we can remove the check here?

Reply:
Yes. We don't need it

Reply:
Removed the check.